modules/apps/blogs/blogs-web/src/main/java/com/liferay/blogs/web/internal/display/context/BlogEntriesDisplayContext.java

```diff
@@ -348,12 +343,9 @@ else if (Objects.equals(orderByCol, "relevance")) {
 		}
 	}
 
-	private Optional<BlogsEntry> _toBlogsEntryOptional(
-		SearchResult searchResult) {
-
+	private BlogsEntry _toBlogsEntry(SearchResult searchResult) {
```

Comment:
@LisztLiszt The original method returns an `Optional` which may or may not contain the real value. That's why the `Stream` above uses `filter(Optional::isPresent)` to only collect the `Optional` instances that have value.

After you change it, you should only add the returned `BlogEntry` to the list when it's not null.